### ISSUE TITLE
Change accessible buffer command navigation keybinding for screen reader users

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#186678]
From a screen reader user's perspective, the terminal accessible buffer is an editor and should behave as such. 

`Ctrl/Cmd+Up/DownArrow` should jump to the top and bottom line. 

We should keep it as is to align with terminals for non screen reader users and provide a new command navigation keybinding for screen reader users.

cc @jooyoungseo

**Comment by jooyoungseo** [comment]
I think this is a Mac-specific issue. CMD+Up/Down/Left/Right arrows are used for text area as a OSX keys to go to very top/bottom of the current document and beginning/ending characters on the current line. These keys are widely adopted in text editing context on Mac, including Text Edit, Mail, and even Monaco editor.

for Windows, Ctrl+Home/Ctrl+End/Home/End are used so it does not have any key conflicts.

To resolve this issue, I suggest VSCode team reassign Alt+Up/DownArrow instead of Ctrl/CMD+Up/Down for shell command navigation.

I tried CMD+Home and CMD+End keys on Mac inside the a11y terminal buffer; however, pressing these keys moves my focus outside the buffer area. In sum, there is no good way for Mac screen reader users to quickly jump to the top/bottom of the terminal output buffer.

**Comment by meganrogge** [comment]
To verify:

- toggle screen reader mode
- create a terminal 
- open the accessibility help menu via `alt+f1` to identify the keybindings for `open detected link` and `go to command`
- then press `Shift+tab` to enter the accessible buffer 
- use these keybindings and verify the commands run as expected
-

**Comment by roblourens** [comment]
I don't totally follow, when in screen reader mode, the commands still say cmd+up/down, and they move the cursor to the top and bottom of the editor.

<img width="649" alt="image" src="https://github.com/microsoft/vscode/assets/323878/e1f72a16-ca47-4cf9-832d-37ecae35b681">

Plus, when screenreader mode is enabled, those commands don't work anymore when _not_in the accessible buffer. Also, I have to leave the accessible buffer to find out what the keybindings are, because they don't show up in the alt+F1 help menu from inside the accessible buffer.

**Comment by Tyriar** [comment]
It's pretty confusing having different keybindings depending on a11y mode. @roblourens may be seeing non-a11y mode keybindings in the help?

**Comment by meganrogge** [comment]
Here's what I see - with or without screen reader mode enabled 

![Screenshot 2023-07-26 at 8 37 44 AM](https://github.com/microsoft/vscode/assets/29464607/aee2568b-17e0-421e-8ffc-c2eb7853ece4)

**Comment by roblourens** [comment]
I haven't customized those. 
<img width="1316" alt="image" src="https://github.com/microsoft/vscode/assets/323878/329a3f80-d0b8-4e29-890b-244ff0da2a1d">

Without screen reader mode on, I see this, and oddly "Go to previous command" changed to "option+up" which is not correct, "cmd+up" still runs that command
<img width="603" alt="image" src="https://github.com/microsoft/vscode/assets/323878/b49cbd4e-3c82-4967-9d23-54e4f094f095">

**Comment by roblourens** [comment]
Another thing I noticed, `To configure the application to be optimized for usage with a Screen Reader press Command+E now.` I press cmd+E, nothing happens

**Comment by meganrogge** [comment]
are you on a special keyboard? I'm pretty confused about what is happening for you

**Comment by roblourens** [comment]
Unplugged the external keyboard, same result

<img width="830" alt="image" src="https://github.com/microsoft/vscode/assets/323878/470fdb84-a9cd-4699-a52f-fe49520d60cc">

**Comment by meganrogge** [comment]
So now it shows the correct binding for go to next/previous command and neither of the commands of interest have keybindings 🤔 . 

Since it shows in the keybindings editor for you - does using the keybinding work? For example cmd + shift + g with focus in the accessible buffer?

**Comment by meganrogge** [comment]
btw I had noticed this earlier, so have an issue tracking the help menu for the accessible buffer being wrong https://github.com/microsoft/vscode/issues/188937

**Comment by roblourens** [comment]
> So now it shows the correct binding for go to next/previous command 

No, it's showing option, but the correct keybindings are with command

Ah, I know what's up with the cmd+shift+G keybinding at least. I have a custom keybinding for that one to open the SCM view. When I remove that, it shows up for "Navigate Accessible Buffer". So that is probably correct behavior at least

I commented out my whole keybindings file

<img width="842" alt="image" src="https://github.com/microsoft/vscode/assets/323878/e9dcbc76-f507-48d7-bfdc-efd59acecd43">

Now that and "Open Detected Link" show up but "Go To Next/Previous" are still wrong

**Comment by meganrogge** [comment]
I think I understand the issue now. When you have screen reader mode enabled, `Option` is used to navigate commands in the accessible buffer. That's not mentioned in this help menu and is a bug. 

The expectation is that SR users won't use the Cmd + Up/DownArrow feature in the terminal because the terminal is not accessible.

**Comment by meganrogge** [comment]
It says `Command Up/DownArrow` for me here. I am not sure why it's saying `Option` for you - do you have an accessible buffer open when you invoke this help menu?

![Screenshot 2023-07-26 at 1 44 08 PM](https://github.com/microsoft/vscode/assets/29464607/87dba474-f3ab-4782-9a0e-935067b06630)

**Comment by meganrogge** [comment]
It uses `Option` instead of `Command` on mac in screen reader mode when one is within the accessible buffer. That should be added to this menu.

**Comment by meganrogge** [comment]
the second one is applied so for you `terminalAccessibleBufferFocus` must be true. can you pls inspect the context keys to confirm that theory?

![Screenshot 2023-07-26 at 1 46 43 PM](https://github.com/microsoft/vscode/assets/29464607/82602081-6176-4cd2-aa44-41c3bf51c4d4)

**Comment by roblourens** [comment]
I don't see that context key 
<img width="324" alt="image" src="https://github.com/microsoft/vscode/assets/323878/f1f00467-0bd1-467e-b055-9a88a7d7693c">

**Comment by roblourens** [comment]
I assuem you hit this case, and the code assumes the keybindings come in a certain order. I don't see why that would be safe https://github.com/microsoft/vscode/blob/5b4a3822f05ed897c6bf3fc02a9c478a734ac0b1/src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibilityHelp.ts#L59-L61
